utils/writeable/src/try_writeable.rs

```diff
@@ -311,10 +315,10 @@ where
     }
 
     #[inline]
-    fn try_write_to_string(&self) -> Result<Cow<str>, Self::Error> {
+    fn try_write_to_string(&self) -> Result<Cow<str>, (Self::Error, Cow<str>)> {
         match self {
             Ok(t) => Ok(t.write_to_string()),
-            Err(e) => Err(e.clone()),
+            Err(e) => Err((e.clone(), e.write_to_string())),
```

Comment:
Observation: in principle `e.write_to_string()` could return a borrowed string; in many cases it will, since it might be printing a static string. This is behavior that wasn't possible before.